packages/builtin/array.pony

```diff
@@ -276,6 +276,14 @@ class Array[A] is Seq[A]
 
     this
 
+  fun add(that: this->Array[A]): Array[this->A!] =>
```

Comment:
The other add feature, I realized, requires a change to String.add which currently only allows String box whereas String.append allows a ReadSeq. 

As to iso^, I can get that signature to compile. I am not so good with generics. Not sure if its because the signature doesn't make sense or if I am doing something wrong.